deepspeed/accelerator/cuda_accelerator.py

```diff
@@ -0,0 +1,123 @@
+from deepspeed.accelerator.abstract_accelerator import DeepSpeedAccelerator
+import torch.cuda
+
+
+class CUDA_Accelerator(DeepSpeedAccelerator):
+    def __init__(self):
+        self.name = 'cuda'
+        self.communication_backend = 'nccl'
+        self.DoubleTensor = torch.cuda.DoubleTensor
+        self.LongTensor = torch.cuda.LongTensor
+        self.FloatTensor = torch.cuda.FloatTensor
+        self.BFloat16Tensor = torch.cuda.BFloat16Tensor
+        self.HalfTensor = torch.cuda.HalfTensor
+        self.IntTensor = torch.cuda.IntTensor
+        self.ByteTensor = torch.cuda.ByteTensor
+
+    # Device APIs
+    def device(self, device_index=None):
+        return torch.cuda.device(device_index)
+
+    def set_device(self, device_index):
+        torch.cuda.set_device(device_index)
+
+    def current_device(self):
+        return torch.cuda.current_device()
+
+    def device_count(self):
+        return torch.cuda.device_count()
+
+    def synchronize(self, device_index=None):
+        return torch.cuda.synchronize(device_index)
+
+    # RNG APIs
+    def set_rng_state(self, new_state, device_index=None):
+        return torch.cuda.set_rng_state(new_state, device_index)
+
+    def get_rng_state(self, device_index=None):
+        return torch.cuda.get_rng_state(device_index)
```

Comment:
When I try to implement get_rng_state under XPU_Accelerator, it looks like torch.xpu.get_rng_state() cannot accept None as input, from pytorch document (https://pytorch.org/docs/stable/generated/torch.cuda.get_rng_state.html) 'None' does not seem to be a valid argument value.

Reply:
Good catch. I think the `device_index=None` is okay for DeepSpeedAccelerator since specific concrete accelerators may do as they please. For example, I think I can rewrite the CUDA_Accelerator version as 

```python

    def get_rng_state(self, device=None):
          if device is None:
              return torch.cuda.get_rng_state()
         return torch.cuda.get_rng_state(cuda_device)
```

What do you think?

Reply:
Yes, thats also what we implemented in XPU_Accelerator